quic/s2n-quic-core/src/recovery/cubic.rs

```diff
@@ -204,7 +205,9 @@ impl CongestionController for CubicCongestionController {
             .try_add(bytes_sent)
             .expect("bytes sent should not exceed u32::MAX");
 
-        self.under_utilized = self.is_congestion_window_under_utilized();
+        if let Some(app_limited) = app_limited {
+            self.under_utilized = app_limited && self.is_congestion_window_under_utilized();
```

Comment:
Do we need to call is underutilized here? Isn't that what the application space is doing already?

Reply:
`is_congestion_window_under_utilized()` is more lenient than the check in application space, we're only under-utilized if there are more than 3 MTU's of space left in the cwnd. In slow start we only need to use half the cwnd to be utilized. See #474